sds/src/scanner/mod.rs

```diff
@@ -286,6 +315,58 @@ impl Scanner {
         output_rule_matches
     }
 
+    #[cfg(feature = "match_validation")]
+    pub async fn validate_matches(
+        &self,
+        rule_matches: &mut Vec<RuleMatch>,
+    ) -> Result<(), MatchValidationError> {
+        if !self.scanner_features.return_matches {
+            return Err(MatchValidationError::NoMatchValidationType);
+        }
+        // Create MatchValidatorRuleMatch per match_validator_type to pass it to each match_validator
+        let mut match_validator_rule_match_per_type = AHashMap::new();
+        for rule_match in rule_matches.drain(..) {
+            let rule = &self.rules[rule_match.rule_index];
+            if let Some(match_validation_type) = rule.get_match_validation_type() {
+                if !match_validator_rule_match_per_type.contains_key(match_validation_type) {
+                    match_validator_rule_match_per_type.insert(match_validation_type, Vec::new());
+                }
+                match_validator_rule_match_per_type
+                    .get_mut(match_validation_type)
+                    .unwrap()
+                    .push(rule_match);
+            }
+        }
+
+        // Call the validate per match_validator_type with their matches and the RuleMatch list and collect the results
+        let futures = match_validator_rule_match_per_type
+            .iter_mut()
+            .filter_map(|(match_validation_type, matches_per_type)| {
+                let match_validator = self.match_validators_per_type.get(match_validation_type);
+                if let Some(match_validator) = match_validator {
+                    Some(
+                        match_validator
+                            .as_ref()
+                            .validate(matches_per_type, &self.rules),
+                    )
+                } else {
+                    None
+                }
```

Comment:
I think here you can simplify the code by using map:
```suggestion
                let match_validator = self.match_validators_per_type.get(match_validation_type).map(|match_validator| match_validator.as_ref().validate(matches_per_type, &self.rules))
                }
```

Reply:
nope it does not work and filter_map expect an Option in return so it can do both map and filter (filtering None validtor)